### ISSUE TITLE
Spork Data Bag Creation - Missing Secondary Object

### ISSUE THREAD

**Issue by TheSerapher** [jonlives/knife-spork#156]
I have been playing around with adding various git plugins for the different asks (env create/edit/delete) and recently started to do the same for data bags. When trying to add a system that worked for environments into the data bag API I noticed these are not working properly due to an issue with addressing them:

```
knife spork data bag create test test3
Created data_bag_item[test3]
 {
+  "test": "https://172.23.42.9:443/data/test/test",
+  "test2": "https://172.23.42.9:443/data/test/test2",
+  "test3": "https://172.23.42.9:443/data/test/test3"
 }
\ No newline at end of file
```

In my plugin, I am simply doing this:

```
      def after_databagcreate
        puts object_difference
      end
```

So it seems that [create_databag](https://github.com/jonlives/knife-spork/blob/08933d7e9245023d4df5ff870adb702c23ddd43f/lib/chef/knife/spork-databag-create.rb#L37) works as intended but when fetching the diff it does not take the secondary object (the actual data bag item) into account. Hence it returns all data for the data BAG test instead of the actual item.

I am guessing that both [these](https://github.com/jonlives/knife-spork/blob/08933d7e9245023d4df5ff870adb702c23ddd43f/lib/knife-spork/runner.rb#L48-L49) need to be set in order for this to work and allowing other plugins to access the data bag BAG and ITEM variables.

**Comment by jonlives** [comment]
@TheSerapher a fix for this has been pushed to the 1.5.0 branch and will be included with the next release once all other PRs have been merged and tested etc  - good catch BTW, you were totally right about the cause of this. I totally missed it!

**Comment by TheSerapher** [comment]
Lucky hunch ;-) And thanks for pushing a fix for the future release!
